Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Plan.hs

```diff
@@ -287,31 +286,29 @@ createMandateInvoiceAndOrder :: Id SP.Person -> Id DM.Merchant -> Plan -> Flow (
 createMandateInvoiceAndOrder driverId merchantId plan = do
   driverPendingAndDuesFees <- QDF.findAllPendingAndDueDriverFeeByDriverId driverId
   driverRegisterationFee <- QDF.findLatestRegisterationFeeByDriverId (cast driverId)
-  allPlans <- QPD.fetchAllPlan
-  let allPlansMaxAmount = List.maximum $ allPlans <&> (.maxAmount)
   let currentDues = sum $ map (\dueInvoice -> fromIntegral dueInvoice.govtCharges + fromIntegral dueInvoice.platformFee.fee + dueInvoice.platformFee.cgst + dueInvoice.platformFee.sgst) driverPendingAndDuesFees
   case driverRegisterationFee of
     Just registerFee -> do
       invoice <- QINV.findByDriverFeeIdAndActiveStatus registerFee.id
       case invoice of
-        Just inv -> SPayment.createOrder (driverId, merchantId) (registerFee : driverPendingAndDuesFees) (Just $ mandateOrder currentDues allPlansMaxAmount) (Just (inv.id, inv.invoiceShortId))
-        Nothing -> createOrderForDriverFee driverPendingAndDuesFees registerFee currentDues allPlansMaxAmount
+        Just inv -> SPayment.createOrder (driverId, merchantId) (registerFee : driverPendingAndDuesFees) (Just $ mandateOrder currentDues) (Just (inv.id, inv.invoiceShortId))
+        Nothing -> createOrderForDriverFee driverPendingAndDuesFees registerFee currentDues
     Nothing -> do
       driverFee <- mkDriverFee
       QDF.create driverFee
-      createOrderForDriverFee driverPendingAndDuesFees driverFee currentDues allPlansMaxAmount
+      createOrderForDriverFee driverPendingAndDuesFees driverFee currentDues
   where
-    mandateOrder currentDues allPlansMaxAmount =
+    mandateOrder currentDues =
       SPayment.MandateOrder
-        { maxAmount = max allPlansMaxAmount currentDues,
+        { maxAmount = max plan.maxAmount currentDues,
```

Comment:
Shouldn't this be `plan.maxAmount`? If we allow credit, dues might exceed `maxAmount` but we still want `maxAmount` only, right?

Reply:
We want to show the amount that is there in the screen to the drivers on psp_page so that they don't feel cheated so if there are dues then dues should be shown in max amount.